frontend/passkeys.js

```diff
@@ -0,0 +1,35 @@
+document.addEventListener('DOMContentLoaded', function() {
+    'use strict';
+
+    const registerPasskeyButtons = document.querySelectorAll('#register-passkey');
+
+
+    async function registerPasskey() {
+        const response = await fetch('/passkey/registration-options') //fetch PublicKeyCredentialCreationOptions as JSON string
+        const publicKeyCredentialCreationOptionsJSON = await response.json() // convert to JSONObject
+        const credentialCreationOptions = PublicKeyCredential.parseCreationOptionsFromJSON(publicKeyCredentialCreationOptionsJSON); // convert to PublicKeyCredentialCreationOptions
+        const publicKeyCredential = await navigator.credentials.create({publicKey: credentialCreationOptions}); // create PublicKeyCredential
+        console.log('publicKeyCredential: ', publicKeyCredential);
+        const registrationResponseJSON = publicKeyCredential.toJSON(); // convert to JSONObject
+        await fetch('/passkey/register', {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'text/plain',
+                'Csrf-Token': `${registerPasskeyButtons[0].getAttribute('csrf-token')}`
+            },
+            body: JSON.stringify(registrationResponseJSON)
+        });
+    }
+
+    registerPasskeyButtons.forEach(function(btn) {
+        btn.addEventListener('click', function(e) {
+            e.preventDefault();
+            registerPasskey().then(function() {
+                console.log('Registration successful');
+            }).catch(function(err) {
+                console.error('Error during registration:', err);
```

Comment:
I wonder whether this logging adds any value to just looking at the network requests.  I think we also saw that a 403 was appearing as a success so it might even be misleading.

Reply:
I did get an error during registration when working locally, so I think it might be useful to keep it. It highlighted a type error.